src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.models import RepositoryProjectPathConfig
+from sentry.shared_integrations.exceptions import ApiError
+
+
+def get_link(config, filepath, version):
+    # feel like there has got to be a better way to do this
+    # if we already have the oi
+    oi = config.organization_integration
+    integration = oi.integration
+    install = integration.get_installation(oi.organization_id)
+
+    filepath.replace(config.stack_root, config.source_root)
+
+    return install.get_stacktrace_link(config.repository, filepath, version)
+
+
+class ProjectStacktraceLinkEndpoint(ProjectEndpoint):
+    """
+        Returns valid links for source code providers so that
+        users can go from the file in the stack trace to the
+        provider of their choice.
+
+        `filepath`: The file path from the strack trace
+        `commitId` (optional): The commit_id for the last commit of the
+                               release associated to the stack trace's event
+
+    """
+
+    def get(self, request, project):
+        # should probably feature gate
+        filepath = request.GET.get("file")
+        commitId = request.GET.get("commitId")
+
+        results = []
+        # try and find any configurations set up, if there
+        # aren't any then we don't do anything else
+        configs = RepositoryProjectPathConfig.objects.filter(project=project,)
+
+        if not configs:
+            return Response(results)
+
+        for config in configs:
+            version = commitId or config.default_branch
+            try:
+                link = get_link(config, filepath, version)
+                # don't think we need the whole configuration, but we
+                # may want to know what provider it is e.g. GH
+                results.append({"config": config, "source_url": link})
+
+            except ApiError:
+                # this could mean either we couldn't find a match
```

Comment:
@MeredithAnya I think we need to return an error to the browser if there is an `ApiError` other than the expected ones for a file not being found (which we catch in `get_stacktrace_link`). We can't really ignore the error and move on here IMO.

Reply:
@scefali yeah I agree. I'm not sure whether we should return a specific message or just a status code and have the frontend render a general error message. I can return a 400 for now? doesn't feel like that's the best fit but not sure what else is better

Reply:
@MeredithAnya I think it's fine to just not even have a try/catch here and let any API error generate a 500 error which will hit Sentry. Unless there is a bug, these errors should be relatively infrequent (I hope).

Reply:
@scefali I just don't want the stack trace to not render just because an external service fails, maybe the error boundaries in the frontend will handle that?

Reply:
@MeredithAnya Yea, I think the front end should handle that. Since we could have other types of errors as well.